src/boot/grub2/ostree-grub-generator

```diff
@@ -61,7 +61,7 @@ read_config()
 
 populate_menu()
 {
-    # Default to /boot if OSTREE_BOOT_PARTITION is not set and /boot and / are on the save device
+    # Default to /boot if OSTREE_BOOT_PARTITION is not set and /boot and / are on the same device
     if [ -z ${OSTREE_BOOT_PARTITION+x} ] && [ $(df / /boot|awk '{print $1}'|uniq|wc -l) -eq 2 ]; then
```

Comment:
How about comparing `stat -c '%d %i'`?

But that said I think I'd prefer to pass this down by env from the upgrader which already determined this, see the `is_ro_mount()` bits - we'd just need to record whether we found a mount or not, in addition to whether it's read-only.

(But I'd also merge the patch if we used the `stat` too)

Reply:
The inode number is different in my use case:
```
root@genericx86-64:~# stat -c '%d %i' / /boot
2050 68658
2050 22
```
Will you be ok with just comparing the device number (%d)?

Reply:
Hmm...but bind mounts should have the same underlying device/inode, right?

Reply:
The bind-mount on ```/boot``` points to the inode of ```/sysroot/boot``` which is different from the inode of ```/```

```
root@genericx86-64:~# stat -c '%d %i' /boot /sysroot/boot/            
2050 22
2050 22
```

And I'm not sure if comparing against ```/sysroot/boot```rather than / is a good idea